src/oscar/apps/search/search_handlers.py

```diff
@@ -9,7 +9,43 @@
 FacetMunger = get_class('search.facets', 'FacetMunger')
 
 
-class SearchHandler(object):
+class SearchResultsPaginationMixin(object):
+    paginate_by = None
+    paginator_class = Paginator
+    page_kwarg = 'page'
+
+    def get_paginate_by(self, *args, **kwargs):
+        return self.paginate_by
+
+    def paginate_queryset(self, queryset, page_size):
+        """
+        Paginate the search results. This is a simplified version of
+        Django's MultipleObjectMixin.paginate_queryset
+        """
+        paginator = self.get_paginator(queryset, page_size)
+        page_kwarg = self.page_kwarg
+        page_number = self.request_data.get(page_kwarg, 1)
+        try:
+            page_number = int(page_number)
+        except ValueError:
+            if page_number == 'last':
+                page_number = paginator.num_pages
+            else:
+                raise InvalidPage(_(
```

Comment:
This exception message is never displayed to the user, so we can probably do without it being translated.

Reply:
Yep, agreed. Done.